src/components/Add/AddPost.tsx

```diff
@@ -0,0 +1,381 @@
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+  
+      if (response.ok) {
+        const responseData = await response.json();
+        dispatch(postSlice.actions.addPost(responseData));
+        // Czyszczenie stanów po pomyślnym dodaniu postu
+        setTitle('');
+        setContent('');
+        setTags([]);
+        setUploadedImage(null);
+        setCategory('');
+        // Ustawienie błędów na null
+        setTitleError(null);
+        setContentError(null);
+        setTagsError(null);
+        setImageError(null);
+        setCategoryError(null);
+      } else {
+        console.error('Failed to add post');
+      }
+    } catch (error) {
+      console.error('Error adding post:', error);
+    }
+  };
+  
+
+  const validateTitle = (value: string) => {
+    try {
+      schema.pick({ title: true }).parse({ title: value });
+      setTitleError(null);
+    } catch (err: unknown) {
+      if (err instanceof ZodError) {
+        setTitleError(err.errors[0].message);
+      }
+    }
+  };
+
+  const validateContent = (value: string) => {
+    try {
+      schema.pick({ content: true }).parse({ content: value });
+      setContentError(null);
+    } catch (err: unknown) {
+      if (err instanceof ZodError) {
+        setContentError(err.errors[0].message);
+      }
+    }
+  };
+
+  const validateTags = (value: string[]) => {
+    try {
+      schema.pick({ tags: true }).parse({ tags: value });
+      setTagsError(null);
+    } catch (err: unknown) {
+      if (err instanceof ZodError) {
+        setTagsError(err.errors[0].message);
+      }
+    }
+  };
+
+  const validateImage = (value: File | null) => {
+    if (!value) {
+      setImageError('Please upload an image.');
+    } else {
+      setImageError(null);
+    }
+  };
+
+  const validateCategory = (value: string) => {
+    try {
+      schema.pick({ category: true }).parse({ category: value });
+      setCategoryError(null);
+    } catch (err: unknown) {
+      if (err instanceof ZodError) {
+        setCategoryError(err.errors[0].message);
+      }
+    }
+  };
+
+  const validateForm = () => {
+    validateTitle(title);
+    validateContent(content);
+    validateTags(tags);
+    validateImage(uploadedImage);
+    validateCategory(category);
+  
+    setContentError(null);
+  
+    let formIsValid = true;
+  
+    if (!titleError && !tagsError && !imageError) {
+      if (!uploadedImage) {
+        setImageError('Please upload an image.');
+        formIsValid = false;
+      }
+  
+      if (!category) {
+        setCategoryError('Please select a category.');
+        formIsValid = false;
+      }
+  
+      if (!content || content.trim().length < 40) {
+        setContentError('Content must be at least 40 characters.');
+        formIsValid = false;
+      }
+    } else {
+      if (content.trim().length < 40) {
+        setContentError('Content must be at least 40 characters.');
```

Comment:
A ty nie masz w zodzie tych wiadomosci o errorze? Pamiętaj ze od tego masz zoda zeby tez wyswietlał informacje o errorach.

Reply:
zod nie ma dostępu do quilla "in real time", bo quill działa asynchronicznie i nie działa to jak w przypadku innych pól w formularzu